torch/_dynamo/convert_frame.py

```diff
@@ -535,7 +535,12 @@ def __call__(
             return ConvertFrameReturn()
 
         if is_generator(code):
-            unimplemented("generator")
+            unimplemented_v2(
+                gb_type="Attempt to trace generator",
+                context="",
+                explanation="Dynamo cannot trace generators as a top-level function.",
```

Comment:
"top-level function" a bit confusion. How about "Don't torch.compile a generator function directly (it is OK to torch.compile a function that calls a generator)"